pkg/reconciler/pipelinerun/resources/apply_test.go

```diff
@@ -1079,6 +1079,66 @@ func TestApplyTaskResultsToPipelineResults(t *testing.T) {
 		runResults  map[string][]v1alpha1.RunResult
 		expected    []v1beta1.PipelineRunResult
 	}{{
+		description: "apply-array-results",
+		results: []v1beta1.PipelineResult{{
+			Name:  "pipeline-result-1",
+			Value: *v1beta1.NewArrayOrString("$(tasks.pt1.results.foo[*])"),
+		}},
+		taskResults: map[string][]v1beta1.TaskRunResult{
+			"pt1": {
+				{
+					Name:  "foo",
+					Value: *v1beta1.NewArrayOrString("do", "rae", "mi"),
+				},
+			},
+		},
+		expected: []v1beta1.PipelineRunResult{{
+			Name:  "pipeline-result-1",
+			Value: *v1beta1.NewArrayOrString("do", "rae", "mi"),
+		}},
+	}, {
+		description: "apply-array-indexing-results",
+		results: []v1beta1.PipelineResult{{
+			Name:  "pipeline-result-1",
+			Value: *v1beta1.NewArrayOrString("$(tasks.pt1.results.foo[1])"),
+		}},
+		taskResults: map[string][]v1beta1.TaskRunResult{
+			"pt1": {
+				{
+					Name:  "foo",
+					Value: *v1beta1.NewArrayOrString("do", "rae", "mi"),
+				},
+			},
+		},
+		expected: []v1beta1.PipelineRunResult{{
+			Name:  "pipeline-result-1",
+			Value: *v1beta1.NewArrayOrString("rae"),
+		}},
+	}, {
+		description: "multiple-array-results-multiple-successful-tasks ",
```

Comment:
a bit confused, this test case is described as "multiple successful tasks" but only one pipeline task is referenced